src/engraving/libmscore/tie.cpp

```diff
@@ -494,17 +497,66 @@ void Tie::slurPos(SlurPos* sp)
     sp->p1    = sc->pos() + sc->segment()->pos() + sc->measure()->pos();
 
     //------p1
-    if ((sc->notes().size() > 1) || (sc->stem() && (sc->up() == _up))) {
-        xo = startNote()->x() + hw * 1.12;
-        yo = startNote()->pos().y() + yOffInside;
+    if (sc->notes().size() > 1) {
+        y1 = startNote()->pos().y() + yOffInside;
+        x1 = 0;
+        qreal xo = startNote()->pos().x() + hw;  // x offset for p1 will be at least note head width
+
+        // ADJUST FOR COLLISIONS ----------------
+
+        for (auto note : sc->notes()) {
+            // adjust for dots
+            if (startNote()->dots().size() > 0) {
+                qreal dotY = note->pos().y() + note->dots().last()->pos().y();
+                if (qAbs(y1 - dotY) < _spatium * 0.5) {
+                    xo = qMax(xo, startNote()->dots().last()->x() + startNote()->dots().last()->width());
+                }
+            }
+
+            // adjust for note collisions
+            if (note == startNote()) {
+                continue;
+            }
+            qreal noteTop = note->y();
+            qreal noteHeight = note->height();
+            if (y1 > noteTop && y1 < noteTop + noteHeight) {
+                xo = qMax(xo, note->x() + note->width());
+            }
+        }
+
+        // adjust for flags (hooks)
+        if (sc->hook() && sc->up()) {
+            if (y1 < sc->hook()->pos().y() + sc->hook()->bbox().height() + (offsetMargin * 2)) {
+                xo = qMax(xo, sc->hook()->pos().x() + sc->hook()->bbox().width());
+            }
+        }
+
+        // adjust for ledger lines
+        if (startNote()->line() >= 8 || startNote()->line() <= 0) {
```

Comment:
This assumes that the notes are on a normal 5 line staff. You actually need check `sc->staff()->staffTypeForElement(sc)->lines()` to get the real number of lines.